### ISSUE TITLE
Unable to move TalkBack cursor on Modal component

### ISSUE THREAD

**Issue by warsawgentleman** [adobe/react-spectrum#7471]
### Provide a general summary of the issue here

When Modal is opened via TalkBack I am unable to change cursor position by swiping left or right. The cursor always comes back to the first focusable element. The flow works perfectly on VoiceOver (macOS).

### 🤔 Expected Behavior?

I should be able to navigate the elements inside Modal by swiping left or right.

### 😯 Current Behavior

TalkBack cursor is blocked on the first element. 

### 🖥️ Steps to Reproduce

The bug is visible in Modal's documentation - https://react-spectrum.adobe.com/react-aria/Modal.html.

1. Turn on TalkBack
2. Go to https://react-spectrum.adobe.com/react-aria/Modal.html
3. Select "Sign up..." button and double tap to open it
4. The first input is selected
5. Swipe right to select the next input
6. TalkBack cursor isn't changed

### Version

1.5.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Android 14. TalkBack 14.1, Chrome

**Comment by LFDanLu** [comment]
Thanks for reporting this, verified locally, digging to see when it started happening.

**Comment by LFDanLu** [comment]
Looks to be something in our FocusScope: https://github.com/adobe/react-spectrum/blob/7eae25e124a9d235ff30f12bd86383b894a1ea04/packages/%40react-aria/focus/src/FocusScope.tsx#L383-L385
where the `activeElement` is now returning the document body after the RAF instead of the newly focused input element after the swipe operation. Replacing this with `e.relatedTarget` seems to work just fine, I can't seem to reproduce the original issue in Firefox even with the updated change

**Comment by LFDanLu** [comment]
The fix above actually breaks another use case of ours since we still do want to coerce focus back to the Modal if focus is genuinely lost to the body but there isn't a great to differentiate that any longer for Talkback it seems. I've filed a bug against Chrome via https://issuetracker.google.com/issues/384844019 to see if they can shed any light as to what may have changed as of late, currently digging to see if there are any alternative paths here.
